### PR TITLE
Update versions for sqlite and protobuf

### PR DESCRIPTION
**Pull request by ahsan-ca**
Updated versions for sqlite and protobuf are needed to fix
security vulnerabilities.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -21,12 +21,12 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 # THE SOFTWARE.
 #####################################################################################
-google/protobuf@v3.11.0 -DCMAKE_POSITION_INDEPENDENT_CODE=On -X subdir -Dprotobuf_BUILD_TESTS=Off
+google/protobuf@v3.20.2 -DCMAKE_POSITION_INDEPENDENT_CODE=On -X subdir -Dprotobuf_BUILD_TESTS=Off
```
Comment (umangyadav) [inline requirements.txt:24]: I wonder if `.pb` files or `.onnx` needs to be re-generated. 
Did all tests pass ?

Comment (umangyadav) [reply requirements.txt:24]: that error is from CK. 
You need to rebuild your dependencies using `rbuild develop` or `rbuild prepare`.

Comment (ahsan-ca) [reply requirements.txt:24]: I tried `rbuild develop` but I still saw the same error. The I tried `rbuild prepare` with the command ` rbuild prepare --deps-dir ../AMDMIGraphX/`. But I still see the same error. I am doing this in the build directory in a docker container. Do you know what I might be missing? :)

Comment (ahsan-ca) [reply requirements.txt:24]: Never mind, I think I got it fixed. Doing `rbuild prepare -d depend` and using `-DCMAKE_PREFIX_PATH=depend` seems to have done the trick.

**Comment by ahsan-ca** [comment]
Seeing this error now:
```
RuntimeError: /code/AMDMIGraphX/AMDMIGraphX/src/targets/gpu/hip.cpp:67: get_available_gpu_memory: Failed getting available memory: invalid argument
```
Not related to this PR.

**Comment by ahsan-ca** [comment]
The get_available_gpu_memory error seem to be machine dependant. I was trying this on another machine and I do not see these errors.
However, I see the following errors. But these errors also appear on develop branch, so do not seem that this PR caused the errors.

```
test_verify: /usr/local/cget/build/tmp-82ea508a033d44d9b28ff38af027eb7d/rocMLIR-507bb94ce7873786486d296ec81d2eadaab49003/external/llvm-project/mlir/lib/Dialect/Tensor/IR/TensorOps.cpp:1394: static RankedTensorType mlir::tensor::CollapseShapeOp::inferCollapsedType(RankedTensorType, ArrayRef<AffineMap>): Assertion `isReassociationValid(reassociation) && "invalid reassociation"' failed.
CMake Error at gdb/test_test_verify_general/run.cmake:16 (message):
  Test failed

[   RUN    ] gemm_2args_mm_5
Benchmarking gpu::mlir_op: 21 configs
Fastest solution: 64,64,32,4,4,2
FAILED: gpu
RMS Error: 0.504296
Max diff: 1.61328
Mismatch at 24: 0.121094 != -0.171875

module: "main"
2 = @param:2 -> float_type, {2, 3, 3, 4}, {36, 12, 4, 1}, target_id=0
1 = @param:1 -> float_type, {2, 1, 2, 3}, {6, 6, 3, 1}, target_id=0
@2 = multibroadcast[out_lens={2, 3, 2, 3},out_dyn_dims={}](1) -> float_type, {2, 3, 2, 3}, {6, 0, 3, 1}, target_id=0
@3 = dot(@2,2) -> float_type, {2, 3, 2, 4}, {24, 8, 4, 1}, target_id=0


ref:
module: "main"
2 = @param:2 -> float_type, {2, 3, 3, 4}, {36, 12, 4, 1}, target_id=0
1 = @param:1 -> float_type, {2, 1, 2, 3}, {6, 6, 3, 1}, target_id=0
@2 = multibroadcast[out_lens={2, 3, 2, 3},out_dyn_dims={}](1) -> float_type, {2, 3, 2, 3}, {6, 0, 3, 1}, target_id=0
@3 = dot(@2,2) -> float_type, {2, 3, 2, 4}, {24, 8, 4, 1}, target_id=0


gpu:
module: "main"
@0 = check_context::migraphx::gpu::context -> float_type, {}, {}, target_id=0
output = @param:output -> float_type, {2, 3, 2, 4}, {24, 8, 4, 1}, target_id=0
2 = @param:2 -> float_type, {2, 3, 3, 4}, {36, 12, 4, 1}, target_id=0
1 = @param:1 -> float_type, {2, 1, 2, 3}, {6, 6, 3, 1}, target_id=0
@4 = multibroadcast[out_lens={2, 3, 2, 3},out_dyn_dims={}](1) -> float_type, {2, 3, 2, 3}, {6, 0, 3, 1}, target_id=0
@5 = gpu::code_object[code_object=3752,symbol_name=mlir_dot,global=384,local=64,](@4,2,output) -> float_type, {2, 3, 2, 4}, {24, 8, 4, 1}, target_id=0



void run_verify::verify(const std::string &, const migraphx::program &, const migraphx::compile_options &) const
/code/AMDMIGraphX/AMDMIGraphX/test/verify/run_verify.cpp:264:
    FAILED: passed [ 0 ]
[  FAILED  ] gemm_2args_mm_5: Test failure
[   RUN    ] gemm_2args_mm_6
Benchmarking gpu::mlir_op: 21 configs
Fastest solution: 64,64,32,8,4,2
FAILED: gpu
RMS Error: 0.292345
Max diff: 1.59375
Mismatch at 24: 0.582031 != 0

module: "main"
2 = @param:2 -> float_type, {1, 3, 3, 4}, {36, 12, 4, 1}, target_id=0
1 = @param:1 -> float_type, {2, 1, 2, 3}, {6, 6, 3, 1}, target_id=0
@2 = multibroadcast[out_lens={2, 3, 2, 3},out_dyn_dims={}](1) -> float_type, {2, 3, 2, 3}, {6, 0, 3, 1}, target_id=0
@3 = multibroadcast[out_lens={2, 3, 3, 4},out_dyn_dims={}](2) -> float_type, {2, 3, 3, 4}, {0, 12, 4, 1}, target_id=0
@4 = dot(@2,@3) -> float_type, {2, 3, 2, 4}, {24, 8, 4, 1}, target_id=0


ref:
module: "main"
2 = @param:2 -> float_type, {1, 3, 3, 4}, {36, 12, 4, 1}, target_id=0
1 = @param:1 -> float_type, {2, 1, 2, 3}, {6, 6, 3, 1}, target_id=0
@2 = multibroadcast[out_lens={2, 3, 2, 3},out_dyn_dims={}](1) -> float_type, {2, 3, 2, 3}, {6, 0, 3, 1}, target_id=0
@3 = multibroadcast[out_lens={2, 3, 3, 4},out_dyn_dims={}](2) -> float_type, {2, 3, 3, 4}, {0, 12, 4, 1}, target_id=0
@4 = dot(@2,@3) -> float_type, {2, 3, 2, 4}, {24, 8, 4, 1}, target_id=0


gpu:
module: "main"
@0 = check_context::migraphx::gpu::context -> float_type, {}, {}, target_id=0
2 = @param:2 -> float_type, {1, 3, 3, 4}, {36, 12, 4, 1}, target_id=0
1 = @param:1 -> float_type, {2, 1, 2, 3}, {6, 6, 3, 1}, target_id=0
@3 = multibroadcast[out_lens={2, 3, 3, 4},out_dyn_dims={}](2) -> float_type, {2, 3, 3, 4}, {0, 12, 4, 1}, target_id=0
@4 = multibroadcast[out_lens={2, 3, 2, 3},out_dyn_dims={}](1) -> float_type, {2, 3, 2, 3}, {6, 0, 3, 1}, target_id=0
output = @param:output -> float_type, {2, 3, 2, 4}, {24, 8, 4, 1}, target_id=0
@6 = gpu::code_object[code_object=3880,symbol_name=mlir_dot,global=384,local=64,](@4,@3,output) -> float_type, {2, 3, 2, 4}, {24, 8, 4, 1}, target_id=0



void run_verify::verify(const std::string &, const migraphx::program &, const migraphx::compile_options &) const
/code/AMDMIGraphX/AMDMIGraphX/test/verify/run_verify.cpp:264:
    FAILED: passed [ 0 ]
[  FAILED  ] gemm_2args_mm_6: Test failure
```

File: requirements.txt (hunk 1 above)
Comment (ahsan-ca) [reply requirements.txt:24]: > I wonder if `.pb` files or `.onnx` needs to be re-generated. Did all tests pass ?

@umangyadav All tests except the test_verify ones I mentioned in the other message pass.

**Comment by umangyadav** [comment]
> However, I see the following errors. But these errors also appear on develop branch, so do not seem that this PR caused the errors.

 They don't seem to be failing on CI with MLIR. Can you check if your MLIR dependency is up-to-date ?

**Comment by krzysz00** [comment]
(Ping noted, no idea what I can do to help here)

**Comment by manupak** [comment]
We dont have a (meaningful) sqlite depedency ( tagging @jerryyin to confirm as jerry worked on it last).
I checked on 7604ecf5b9cd04f9df4219d6f1ef998951ed7449 in develop and I dont see these issues either.

**Comment by manupak** [comment]
I suspect its dev environment issue ? as 1) I cant reproduce it and 2) CI cant reproduce it.. So might worth checking the commit hash being used in the dev env.

**Comment by ahsan-ca** [comment]
> I suspect its dev environment issue ? as 1) I cant reproduce it and 2) CI cant reproduce it.. So might worth checking the MLIR commit hash being used in the dev env.

Thanks for checking it out. I'll investigate it more at my end. I also suspect this to be an env issue at my end.

**Review by pfultz2 — APPROVED** [review]

**Comment by ahsan-ca** [comment]
> The get_available_gpu_memory error seem to be machine dependant. I was trying this on another machine and I do not see these errors. However, I see the following errors. But these errors also appear on develop branch, so do not seem that this PR caused the errors.
> 
> ```
> [   RUN    ] gemm_2args_mm_5
> Benchmarking gpu::mlir_op: 21 configs
> Fastest solution: 64,64,32,4,4,2
> FAILED: gpu
> RMS Error: 0.504296
> Max diff: 1.61328
> Mismatch at 24: 0.121094 != -0.171875
> 
> module: "main"
> 2 = @param:2 -> float_type, {2, 3, 3, 4}, {36, 12, 4, 1}, target_id=0
> 1 = @param:1 -> float_type, {2, 1, 2, 3}, {6, 6, 3, 1}, target_id=0
> @2 = multibroadcast[out_lens={2, 3, 2, 3},out_dyn_dims={}](1) -> float_type, {2, 3, 2, 3}, {6, 0, 3, 1}, target_id=0
> @3 = dot(@2,2) -> float_type, {2, 3, 2, 4}, {24, 8, 4, 1}, target_id=0
> 
> 
> ref:
> module: "main"
> 2 = @param:2 -> float_type, {2, 3, 3, 4}, {36, 12, 4, 1}, target_id=0
> 1 = @param:1 -> float_type, {2, 1, 2, 3}, {6, 6, 3, 1}, target_id=0
> @2 = multibroadcast[out_lens={2, 3, 2, 3},out_dyn_dims={}](1) -> float_type, {2, 3, 2, 3}, {6, 0, 3, 1}, target_id=0
> @3 = dot(@2,2) -> float_type, {2, 3, 2, 4}, {24, 8, 4, 1}, target_id=0
> 
> 
> gpu:
> module: "main"
> @0 = check_context::migraphx::gpu::context -> float_type, {}, {}, target_id=0
> output = @param:output -> float_type, {2, 3, 2, 4}, {24, 8, 4, 1}, target_id=0
> 2 = @param:2 -> float_type, {2, 3, 3, 4}, {36, 12, 4, 1}, target_id=0
> 1 = @param:1 -> float_type, {2, 1, 2, 3}, {6, 6, 3, 1}, target_id=0
> @4 = multibroadcast[out_lens={2, 3, 2, 3},out_dyn_dims={}](1) -> float_type, {2, 3, 2, 3}, {6, 0, 3, 1}, target_id=0
> @5 = gpu::code_object[code_object=3752,symbol_name=mlir_dot,global=384,local=64,](@4,2,output) -> float_type, {2, 3, 2, 4}, {24, 8, 4, 1}, target_id=0
> 
> 
> 
> void run_verify::verify(const std::string &, const migraphx::program &, const migraphx::compile_options &) const
> /code/AMDMIGraphX/AMDMIGraphX/test/verify/run_verify.cpp:264:
>     FAILED: passed [ 0 ]
> [  FAILED  ] gemm_2args_mm_5: Test failure
> [   RUN    ] gemm_2args_mm_6
> Benchmarking gpu::mlir_op: 21 configs
> Fastest solution: 64,64,32,8,4,2
> FAILED: gpu
> RMS Error: 0.292345
> Max diff: 1.59375
> Mismatch at 24: 0.582031 != 0
> 
> module: "main"
> 2 = @param:2 -> float_type, {1, 3, 3, 4}, {36, 12, 4, 1}, target_id=0
> 1 = @param:1 -> float_type, {2, 1, 2, 3}, {6, 6, 3, 1}, target_id=0
> @2 = multibroadcast[out_lens={2, 3, 2, 3},out_dyn_dims={}](1) -> float_type, {2, 3, 2, 3}, {6, 0, 3, 1}, target_id=0
> @3 = multibroadcast[out_lens={2, 3, 3, 4},out_dyn_dims={}](2) -> float_type, {2, 3, 3, 4}, {0, 12, 4, 1}, target_id=0
> @4 = dot(@2,@3) -> float_type, {2, 3, 2, 4}, {24, 8, 4, 1}, target_id=0
> 
> 
> ref:
> module: "main"
> 2 = @param:2 -> float_type, {1, 3, 3, 4}, {36, 12, 4, 1}, target_id=0
> 1 = @param:1 -> float_type, {2, 1, 2, 3}, {6, 6, 3, 1}, target_id=0
> @2 = multibroadcast[out_lens={2, 3, 2, 3},out_dyn_dims={}](1) -> float_type, {2, 3, 2, 3}, {6, 0, 3, 1}, target_id=0
> @3 = multibroadcast[out_lens={2, 3, 3, 4},out_dyn_dims={}](2) -> float_type, {2, 3, 3, 4}, {0, 12, 4, 1}, target_id=0
> @4 = dot(@2,@3) -> float_type, {2, 3, 2, 4}, {24, 8, 4, 1}, target_id=0
> 
> 
> gpu:
> module: "main"
> @0 = check_context::migraphx::gpu::context -> float_type, {}, {}, target_id=0
> 2 = @param:2 -> float_type, {1, 3, 3, 4}, {36, 12, 4, 1}, target_id=0
> 1 = @param:1 -> float_type, {2, 1, 2, 3}, {6, 6, 3, 1}, target_id=0
> @3 = multibroadcast[out_lens={2, 3, 3, 4},out_dyn_dims={}](2) -> float_type, {2, 3, 3, 4}, {0, 12, 4, 1}, target_id=0
> @4 = multibroadcast[out_lens={2, 3, 2, 3},out_dyn_dims={}](1) -> float_type, {2, 3, 2, 3}, {6, 0, 3, 1}, target_id=0
> output = @param:output -> float_type, {2, 3, 2, 4}, {24, 8, 4, 1}, target_id=0
> @6 = gpu::code_object[code_object=3880,symbol_name=mlir_dot,global=384,local=64,](@4,@3,output) -> float_type, {2, 3, 2, 4}, {24, 8, 4, 1}, target_id=0
> 
> 
> 
> void run_verify::verify(const std::string &, const migraphx::program &, const migraphx::compile_options &) const
> /code/AMDMIGraphX/AMDMIGraphX/test/verify/run_verify.cpp:264:
>     FAILED: passed [ 0 ]
> [  FAILED  ] gemm_2args_mm_6: Test failure
> 
> test_verify: /usr/local/cget/build/tmp-82ea508a033d44d9b28ff38af027eb7d/rocMLIR-507bb94ce7873786486d296ec81d2eadaab49003/external/llvm-project/mlir/lib/Dialect/Tensor/IR/TensorOps.cpp:1394: static RankedTensorType mlir::tensor::CollapseShapeOp::inferCollapsedType(RankedTensorType, ArrayRef<AffineMap>): Assertion `isReassociationValid(reassociation) && "invalid reassociation"' failed.
> CMake Error at gdb/test_test_verify_general/run.cmake:16 (message):
>   Test failed
> ```

The failures seem to be happening on Navi32, the tests run successfully on an MI100 system. I have opened an issue for this with the details: https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/issues/2365

FYI @manupak @umangyadav

**Review by umangyadav — APPROVED** [review]

**Review by TedThemistokleous — APPROVED** [review]

**Comment by TedThemistokleous** [comment]
@ahsan-ca once CI passes set this to ready-to-merge unless anyone has any other unresolved comments

**Comment by ahsan-ca** [comment]
> @ahsan-ca once CI passes set this to ready-to-merge unless anyone has any other unresolved comments

@TedThemistokleous There are a few failures, but I am unable to sign in to Jenkins to see the failures. How does one get access to Jenkins to see the failures?

**Comment by causten** [comment]
/opt/rocm/llvm/bin/clang++ -Werror -g -O2 -fno-omit-frame-pointer -fsanitize=undefined,address -fno-sanitize-recover=undefined,address CMakeFiles/test_tf.dir/tf/tf_test.cpp.o -o ../bin/test_tf  -Wl,-rpath,/var/jenkins/workspace/AMDMIGraphX_PR-2347/build/lib ../lib/libmigraphx_tf.so.2008000.0 ../lib/libmigraphx.so.2008000.0 -lpthread 
ld.lld: error: undefined reference due to --no-allow-shlib-undefined: google::protobuf::internal::InternalMetadata::~InternalMetadata()
>>> referenced by ../lib/libmigraphx_tf.so.2008000.0
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [test/CMakeFiles/test_tf.dir/build.make:99: bin/test_tf] Error 1
make[2]: Leaving directory '/var/jenkins/workspace/AMDMIGraphX_PR-2347/build'
make[1]: *** [CMakeFiles/Makefile2:20470: test/CMakeFiles/test_tf.dir/all] E

**Comment by ahsan-ca** [comment]
> /opt/rocm/llvm/bin/clang++ -Werror -g -O2 -fno-omit-frame-pointer -fsanitize=undefined,address -fno-sanitize-recover=undefined,address CMakeFiles/test_tf.dir/tf/tf_test.cpp.o -o ../bin/test_tf -Wl,-rpath,/var/jenkins/workspace/AMDMIGraphX_PR-2347/build/lib ../lib/libmigraphx_tf.so.2008000.0 ../lib/libmigraphx.so.2008000.0 -lpthread ld.lld: error: undefined reference due to --no-allow-shlib-undefined: google::protobuf::internal::InternalMetadata::~InternalMetadata()
> 
> > > > referenced by ../lib/libmigraphx_tf.so.2008000.0
> > > > clang++: error: linker command failed with exit code 1 (use -v to see invocation)
> > > > make[2]: *** [test/CMakeFiles/test_tf.dir/build.make:99: bin/test_tf] Error 1
> > > > make[2]: Leaving directory '/var/jenkins/workspace/AMDMIGraphX_PR-2347/build'
> > > > make[1]: *** [CMakeFiles/Makefile2:20470: test/CMakeFiles/test_tf.dir/all] E

I suspect this to be caused by the issue: [ABI may depend on NDEBUG](https://github.com/protocolbuffers/protobuf/issues/9947). This issue has been fixed in [Protobuf v21.3](https://github.com/protocolbuffers/protobuf/releases/tag/v21.3). I have updated the Protobuf version to v21.3 to see if it fixes the error and how it interacts with other packages. Need to wait for Jenkins tests to see how this plays out.

**Comment by ahsan-ca** [comment]
> > /opt/rocm/llvm/bin/clang++ -Werror -g -O2 -fno-omit-frame-pointer -fsanitize=undefined,address -fno-sanitize-recover=undefined,address CMakeFiles/test_tf.dir/tf/tf_test.cpp.o -o ../bin/test_tf -Wl,-rpath,/var/jenkins/workspace/AMDMIGraphX_PR-2347/build/lib ../lib/libmigraphx_tf.so.2008000.0 ../lib/libmigraphx.so.2008000.0 -lpthread ld.lld: error: undefined reference due to --no-allow-shlib-undefined: google::protobuf::internal::InternalMetadata::~InternalMetadata()
> > > > > referenced by ../lib/libmigraphx_tf.so.2008000.0
> > > > > clang++: error: linker command failed with exit code 1 (use -v to see invocation)
> > > > > make[2]: *** [test/CMakeFiles/test_tf.dir/build.make:99: bin/test_tf] Error 1
> > > > > make[2]: Leaving directory '/var/jenkins/workspace/AMDMIGraphX_PR-2347/build'
> > > > > make[1]: *** [CMakeFiles/Makefile2:20470: test/CMakeFiles/test_tf.dir/all] E
> 
> I suspect this to be caused by an issue in Protobuf : [ABI may depend on NDEBUG](https://github.com/protocolbuffers/protobuf/issues/9947). This issue has been fixed in [Protobuf v21.3](https://github.com/protocolbuffers/protobuf/releases/tag/v21.3). I have updated the Protobuf version to v21.3 to see if it fixes the error and how this version interacts with other packages. Need to wait for Jenkins tests to see how this plays out.

Ah, this did not work out. Looking into it.

**Comment by ahsan-ca** [comment]
> > > /opt/rocm/llvm/bin/clang++ -Werror -g -O2 -fno-omit-frame-pointer -fsanitize=undefined,address -fno-sanitize-recover=undefined,address CMakeFiles/test_tf.dir/tf/tf_test.cpp.o -o ../bin/test_tf -Wl,-rpath,/var/jenkins/workspace/AMDMIGraphX_PR-2347/build/lib ../lib/libmigraphx_tf.so.2008000.0 ../lib/libmigraphx.so.2008000.0 -lpthread ld.lld: error: undefined reference due to --no-allow-shlib-undefined: google::protobuf::internal::InternalMetadata::~InternalMetadata()
> > > > > > referenced by ../lib/libmigraphx_tf.so.2008000.0
> > > > > > clang++: error: linker command failed with exit code 1 (use -v to see invocation)
> > > > > > make[2]: *** [test/CMakeFiles/test_tf.dir/build.make:99: bin/test_tf] Error 1
> > > > > > make[2]: Leaving directory '/var/jenkins/workspace/AMDMIGraphX_PR-2347/build'
> > > > > > make[1]: *** [CMakeFiles/Makefile2:20470: test/CMakeFiles/test_tf.dir/all] E
> > 
> > 
> > I suspect this to be caused by an issue in Protobuf : [ABI may depend on NDEBUG](https://github.com/protocolbuffers/protobuf/issues/9947). This issue has been fixed in [Protobuf v21.3](https://github.com/protocolbuffers/protobuf/releases/tag/v21.3). I have updated the Protobuf version to v21.3 to see if it fixes the error and how this version interacts with other packages. Need to wait for Jenkins tests to see how this plays out.
> 
> Ah, this did not work out. Looking into it.

The `install dependencies` step failed with Protobuf v21.3. I also tried Protobuf v24.3 but it still failed the install dependencies step.

**Comment by ahsan-ca** [comment]
The original version of Protobuf `3.11.0` had the security vulnerability `CVE-2021-22570`. Protobuf version `3.15.0` or greater fixes this vulnerability.
I am now trying Protobuf version `3.19.6` (as Protobuf <= 3.19, most likely, does not have the undefined reference issue) to see if it does not break our CI. If it does not, then we should be good as it satisfies the requirement of fixing security vulnerability `CVE-2021-22570`

**Comment by ahsan-ca** [comment]
> The original version of Protobuf `3.11.0` had the security vulnerability `CVE-2021-22570`. Protobuf version `3.15.0` or greater fixes this vulnerability. I am now trying Protobuf version `3.19.6` (as Protobuf <= 3.19, most likely, does not have the undefined reference issue) to see if it does not break our CI. If it does not, then we should be good as it satisfies the requirement of fixing security vulnerability `CVE-2021-22570`

All the CI tests pass so far, just waiting on Jenkins tests to pass before this is ready to be merged.

**Comment by causten** [comment]
[ 22%] Linking CXX executable ../bin/test_tf

cd /var/jenkins/workspace/AMDMIGraphX_PR-2347/build/test && /usr/local/lib/python3.8/dist-packages/cmake/data/bin/cmake -E cmake_link_script CMakeFiles/test_tf.dir/link.txt --verbose=1

/opt/rocm/llvm/bin/clang++ -Werror -g -O2 -fsanitize=undefined -fno-sanitize=vptr,function -fno-sanitize-recover=undefined CMakeFiles/test_tf.dir/tf/tf_test.cpp.o -o ../bin/test_tf  -Wl,-rpath,/var/jenkins/workspace/AMDMIGraphX_PR-2347/build/lib ../lib/libmigraphx_tf.so.2008000.0 ../lib/libmigraphx.so.2008000.0 -lpthread 

ld.lld: error: undefined reference due to --no-allow-shlib-undefined: google::protobuf::internal::InternalMetadata::~InternalMetadata()

>>> referenced by ../lib/libmigraphx_tf.so.2008000.0

clang++: error: linker command failed with exit code 1 (use -v to see invocation)

make[2]: *** [test/CMakeFiles/test_tf.dir/build.make:99: bin/test_tf] Error 1

make[2]: Leaving directory '/var/jenkins/workspace/AMDMIGraphX_PR-2347/build'

make[1]: *** [CMakeFiles/Makefile2:25531: test/CMakeFiles/test_tf.dir/all] Error 2

make[1]: *** Waiting for unfinished jobs....

**Comment by ahsan-ca** [comment]
> ```
> [ 22%] Linking CXX executable ../bin/test_tf
> cd /var/jenkins/workspace/AMDMIGraphX_PR-2347/build/test && /usr/local/lib/python3.8/dist-packages/cmake/data/bin/cmake -E cmake_link_script CMakeFiles/test_tf.dir/link.txt --verbose=1
> /opt/rocm/llvm/bin/clang++ -Werror -g -O2 -fsanitize=undefined -fno-sanitize=vptr,function -fno-sanitize-recover=undefined CMakeFiles/test_tf.dir/tf/tf_test.cpp.o -o ../bin/test_tf  -Wl,-rpath,/var/jenkins/workspace/AMDMIGraphX_PR-2347/build/lib ../lib/libmigraphx_tf.so.2008000.0 ../lib/libmigraphx.so.2008000.0 -lpthread 
> ld.lld: error: undefined reference due to --no-allow-shlib-undefined: google::protobuf::internal::InternalMetadata::~InternalMetadata()
> >>> referenced by ../lib/libmigraphx_tf.so.2008000.0
> clang++: error: linker command failed with exit code 1 (use -v to see invocation)
> make[2]: *** [test/CMakeFiles/test_tf.dir/build.make:99: bin/test_tf] Error 1
> make[2]: Leaving directory '/var/jenkins/workspace/AMDMIGraphX_PR-2347/build'
> make[1]: *** [CMakeFiles/Makefile2:25531: test/CMakeFiles/test_tf.dir/all] Error 2
> make[1]: *** Waiting for unfinished jobs....
> ```

Ah, the same error persists.

**Comment by ahsan-ca** [comment]
Seems like 3.18.0 worked, I am now trying 3.19.0 in a bid to get the most recent version of Protobuf that does not have the error we saw with later versions of Protobuf.

**Pull request merged**